src/composites.rs

```diff
@@ -0,0 +1,214 @@
+//! Module for combining hyper services
+
+use std::{io, fmt};
+use hyper::server::{Service, NewService};
+use hyper::{Request, Response, StatusCode};
+use futures::{future, Future};
+use context::Context;
+
+/// Trait for getting the path of a request
+pub trait HasPath {
```

Comment:
HasPath and HasNotFound sounds like a bit of a mouthful. What about GetPath? and NotFound?

Reply:
👍